sale_product_set_variant/wizard/product_set_add.py

```diff
@@ -96,12 +110,27 @@ def add_set(self):
         so_lines = []
         for set_line in self.set_line_ids:
             if not set_line.product_variant_ids:
-                raise UserError(_("Please select the "
-                                  "appropriate product variants"))
+                variants = set_line.product_template_id.product_variant_ids
+                if len(variants) == 1:
+                    set_line.product_variant_ids = [(6, 0, variants.ids)]
+                else:
+                    raise UserError(
+                        _("Please select the "
+                          "appropriate product variants "
+                          "for product {}".format(
```

Comment:
The translations `_()` should close right after the closing `"`, here you are translating the text *after* the interpolation.